File: php/src/NBodySimulation/FixedNBodiesSimulation.php
<?php

namespace App\NBodySimulation;

use SplFixedArray;

class FixedNBodiesSimulation
{
    use NBodySimulationTrait;

    protected SplFixedArray $bodies;

    public function __construct(protected readonly int $steps_count)
    {
        $this->bodies = new SplFixedArray(self::BODY_COUNT);

        $this->init();
    }

    public function run(): SplFixedArray
    {
        $this->simulate();

        return $this->bodies;
    }
}

File: c#/Benchmark/Logic/NBodySimulation/FixedNBodiesSimulation.cs
namespace Benchmark.Logic.NBodySimulation;

public class FixedNBodiesSimulation : AbstractNBodySimulation
{
    public FixedNBodiesSimulation(int stepsCount)
    {
        this.Size = stepsCount;
        this.Init();
    }

    protected override Body[] Bodies { get; } = new Body[AbstractNBodySimulation.BODY_COUNT];

    private void Init()
    {
        for (int i = 0; i < AbstractNBodySimulation.BODY_COUNT; i++) {
            this.Bodies[i] = new();
        }
    }

    public IList<Body> Run()
    {
        this.Simulate();

        return this.Bodies;
    }
}
